lib/segment/src/vector_storage/quantized/quantized_query_scorer.rs

```diff
@@ -1,41 +1,45 @@
 use std::borrow::Cow;
-use std::marker::PhantomData;
 
 use common::counter::hardware_counter::HardwareCounterCell;
 use common::types::{PointOffsetType, ScoreType};
 
 use crate::data_types::primitive::PrimitiveVectorElement;
-use crate::data_types::vectors::DenseVector;
+use crate::data_types::vectors::{DenseVector, VectorElementType};
 use crate::spaces::metric::Metric;
 use crate::types::QuantizationConfig;
 use crate::vector_storage::query_scorer::QueryScorer;
 
-pub struct QuantizedQueryScorer<'a, TElement, TMetric, TEncodedVectors>
+pub struct QuantizedQueryScorer<'a, TEncodedVectors>
 where
-    TElement: PrimitiveVectorElement,
-    TMetric: Metric<TElement>,
     TEncodedVectors: quantization::EncodedVectors,
 {
     query: TEncodedVectors::EncodedQuery,
     quantized_data: &'a TEncodedVectors,
-    metric: PhantomData<TMetric>,
-    element: PhantomData<TElement>,
     hardware_counter: HardwareCounterCell,
 }
 
-impl<'a, TElement, TMetric, TEncodedVectors>
-    QuantizedQueryScorer<'a, TElement, TMetric, TEncodedVectors>
+pub enum QuantizedInternalScorerResult<'a, TEncodedVectors>
```

Comment:
Result of internal scorer creation. It can be constructed scorer or nothing but we want to get hardware counter ownership back